### PR TITLE
Update MAGIC.md with new monsters spells examples

### PR DESCRIPTION
**Pull request by Kenan2000**
#### Summary
SUMMARY: None
**Update MAGIC.md with new monster spells examples - includes both basic and advanced examples**

#### Purpose of change

_Expansion of MAGIC.md - to show how monsters spells JSON code can be properly manipulated for new modders_

#### Describe the solution

**_Makes MAGIC.md more thorough and gives proper explanations to very important monster spells JSON code_**

#### Describe alternatives you've considered

_Do not include this into MAGIC.md_

#### Testing

_The code has been tested in-game, it works just fine with no bugs_

#### Additional context

**I would like to** **thank** @Axema  for greatly helping me with editing some parts of the guide to make it more clear, **of course** **thank the developers** for making such interesting JSON system and the game itself 😃

### CODE REVIEW

**Comment by Fris0uman** [comment]
why not follow the format of the rest of the documentation and explain what each field does?

**Comment by Kenan2000** [comment]
> why not follow the format of the rest of the documentation and explain what each field does?

Ah, can you elaborate, please?I may have messed up a little bit, I tried to make it as thorough as I could and only explain the most important bits

**Comment by Kenan2000** [comment]
@Fris0uman updated it to fit the original code format, hope I've done it nicely 😄

File: doc/MAGIC.md [hunk 1]
```diff
@@ -232,6 +232,149 @@ You can add a "spell" member to professions or an NPC class definition like so:
...
+    "type": "SPELL",
+    "name": "Combo Strikes",                                   // name of the spell that shows in game                              
+    "description": "Upon casting this spell, will also activate the spells specified on the 'extra_effects' in descending order.",
+    "flags": [ "SILENT", "RANDOM_DAMAGE", "RANDOM_AOE" ],      // see "Spell Flags" in this document
+    "valid_targets": [ "hostile", "ground" ],                  // if a valid target is not included, you cannot cast the spell on that target. 
+    "effect": "projectile_attack",                             // effects are coded in C++. A list is provided in this document of possible effects that have been coded.
+    "effect_str": "downed",                                    // varies, see table of implemented effects in this document
+    "extra_effects": [ { "id": "test_atk1" }, { "id": "test_atk2" } ],               // this allows you to cast multiple spells with only one spell
+    "min_damage": 7,                                         // minimum damage (or "starting" damage)
+    "max_damage": 14,                                        // maximum  damage the spell can achieve
```
Comment (LaVeyanFiend) [inline doc/MAGIC.md:296]:
```suggestion
    "min_damage": 7,                                         // minimum damage (or "starting" damage)
    "max_damage": 14,                                        // maximum  damage the spell can achieve
    "damage_increment": 0.7                            // damage increase per spell level
```

File: doc/MAGIC.md [hunk 2]
```diff
@@ -232,6 +232,149 @@ You can add a "spell" member to professions or an NPC class definition like so:
...
+    "id": "test_attack_repeat",                         // id of the spell, used internally. not translated
+    "name": "a spell",                                  // name of the spell that shows in game
+    "description": "Upon casting this spell it will repeat the spell specified in `extra_effects` - the amount of repetitions is the interval `min_damage`-`max_damage` ",
+    "extra_effects": [ { "id": "test_attack" } ],       // this allows you to cast multiple spells with only one spell
+    "flags": [ "SILENT", "WONDER", "RANDOM_DAMAGE" ],   // see "Spell Flags" in this document
+    "valid_targets": [ "hostile" ],                     // if a valid target is not included, you cannot cast the spell on that target.
+    "effect": "target_attack",                          // effects are coded in C++. A list is provided in this document of possible effects that have been coded.
+    "effect_str": "target_message",                     // varies, see table of implemented effects in this document
+    "min_damage": 5,                                    // minimum (starting damage)
+    "max_damage": 7,                                    // maximum damage the spell can achieve
```
Comment (LaVeyanFiend) [inline doc/MAGIC.md:348]:
```suggestion
    "min_damage": 5,                                    // minimum (starting damage)
    "max_damage": 7,                                    // maximum damage the spell can achieve
    "damage_increment": 0.2                        // damage increase per spell level
```

**Pull request merged**